### ISSUE TITLE
on_job_stderr: abort() on loop_poll_events()

### ISSUE THREAD

**Issue by justinmk** [neovim/neovim#7707]
(moved from https://github.com/neovim/neovim/issues/3197#issuecomment-350379278 @markstos)

```
 #1  0x00007eff9db4ff5d in __GI_abort () at abort.c:90
 ...
 #2  0x000055d8b708e720 in loop_poll_events ()
 ...
 #3  0x000055d8b7144575 in os_breakcheck ()
 ...
 #5  0x000055d8b70da8c2 in vgetc ()
 ...
 #12 0x000055d8b710d4cd in emsgf ()
 ...
 #13 0x000055d8b70e22a8 in hash_add ()
 ...
 #17 0x000055d8b704069c in on_job_stderr ()
 ...
 #25 0x000055d8b71de928 in uv_run ()
 #26 0x000055d8b708e746 in loop_poll_events ()
 #27 0x000055d8b7144575 in os_breakcheck ()
 ...
 #37 0x000055d8b70706b5 in call_func ()
 ...
 #47 0x000055d8b7075539 in eval0 ()
 ...
 #52 0x000055d8b7070642 in call_func ()
 ...
 #66 0x000055d8b7070642 in call_func ()
```

@bfredl What changed since 20171126 that would avoid this? Channels refactor was merged on 20171126.

**Comment by justinmk** [comment]
Duplicate of https://github.com/neovim/neovim/issues/7699

**Comment by bfredl** [comment]
Not a duplicate, rather fixed by #7646, by deferring that emsg until it's safe.
